core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala

```diff
@@ -488,49 +483,40 @@ object H2OSchemaUtils {
   private def collectSimpleLikeTypes(flatSchema: StructType): Seq[Int] = {
     flatSchema.fields.zipWithIndex.flatMap { case (field, idx) =>
       field.dataType match {
-        case BooleanType => Option(idx)
-        case ByteType => Option(idx)
-        case ShortType => Option(idx)
-        case IntegerType => Option(idx)
-        case LongType => Option(idx)
-        case FloatType => Option(idx)
-        case _: DecimalType => Option(idx)
-        case DoubleType => Option(idx)
-        case StringType => Option(idx)
-        case TimestampType => Option(idx)
-        case DateType => Option(idx)
+        case BooleanType => Some(idx)
+        case ByteType => Some(idx)
+        case ShortType => Some(idx)
+        case IntegerType => Some(idx)
+        case LongType => Some(idx)
+        case FloatType => Some(idx)
+        case _: DecimalType => Some(idx)
+        case DoubleType => Some(idx)
+        case StringType => Some(idx)
+        case TimestampType => Some(idx)
+        case DateType => Some(idx)
         case _ => None
       }
     }
   }
 
-
   /**
-    * Get size of Array or Vector type. Already expects flat DataFrame
+    * Get size of Array or Vector type. Expects already flattened DataFrame
     *
     * @param row current row
     * @param idx index of the element
     */
   private def getCollectionSize(row: Row, idx: Int): Int = {
     if (row.isNullAt(idx)) {
-      return 0
-    }
-
-    val elemType = row.schema.fields(idx)
-    elemType.dataType match {
-      case ArrayType(_, _) => row.getAs[Seq[_]](idx).length
-      case BinaryType => row.getAs[Array[Byte]](idx).length
-      // it is user defined type - currently, only vectors are supported
-      case _ =>
-        val value = row.get(idx)
-        value match {
-          case _: mllib.linalg.Vector =>
-            row.getAs[mllib.linalg.Vector](idx).size
-          case _: ml.linalg.Vector =>
-            row.getAs[ml.linalg.Vector](idx).size
-          case _ =>
-            throw new UnsupportedOperationException(s"User defined type is not supported: ${value.getClass}")
-        }
+      0
+    } else {
+      val dataType = row.schema.fields(idx).dataType
```

Comment:
Just a note: `schema` could be `null`

Reply:
Good catch! Is it possible that row.schema is null even though whole dataFrame has schema defined?